pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -241,7 +241,9 @@ def __repr__(self) -> str:
         return f"{self.__class__.__name__}({state})"
 
     def __getstate__(self) -> dict:
-        d = super().__getstate__()
+        sync_manager = self._apply_sync if self.is_tensor else self.value._apply_sync
```

Comment:
where do these come from and why are they needed?

Reply:
This comes from: https://github.com/PyTorchLightning/metrics/pull/302

Here are the steps to restart on failure
1. Perform sync between processes and accumulate states on rank 0 only as only rank 0 saves a checkpoint. This is done by the `_apply_sync` function added to torchmetrics.
2. Drop metrics reference and any pickled objects. 
3. On reload, each process will reload accumulated states, therefore we need to reset them on rank different than 0.
4. Keep training.